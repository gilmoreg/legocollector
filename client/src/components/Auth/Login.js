import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { connect } from 'react-redux';
import { login, fetchWatches } from '../../state/actions';
import { API_URL } from '../../config';

const StyledButton = styled.button`
  border: none;
  background-color: transparent;
`;

export class Login extends Component {
  constructor(props) {
    super(props);
    this.storeProfile = this.storeProfile.bind(this);
    this.fetchProfile = this.fetchProfile.bind(this);
    this.amazonLogin = this.amazonLogin.bind(this);
  }

  storeProfile({ token, email }) {
    try {
      window.localStorage.setItem(
        'legocollectorProfile',
        JSON.stringify({ token, email }),
      );
    } catch (e) {
      console.error(e);
    }
  };

  fetchProfile(accessToken) {
    return fetch(`${API_URL}/login`,
      { method: 'POST',
        body: JSON.stringify({ access_token: accessToken }),
      })
      .then(res => res.json())
      .then(res => res.result);
  }

  amazonLogin() {
    return window.amazon.Login.authorize({ scope: 'profile' },
      response =>
        this.fetchProfile(response.access_token)
          .then((profile) => {
            this.storeProfile(profile);
            this.props.dispatch(login(profile));
            this.props.dispatch(fetchWatches(profile.token));
          })
          .catch(error => console.error(error)),
      error => console.error(error));
  }

  render() {
    return (
      <StyledButton id="LoginWithAmazon" onClick={this.amazonLogin}>
        <img
          alt="Login with Amazon"
          src="https://images-na.ssl-images-amazon.com/images/G/01/lwa/btnLWA_gold_156x32.png"
          width="156"
          height="32"
        />
      </StyledButton>
    );
  }
}

Login.propTypes = {
  dispatch: PropTypes.func.isRequired,
};

export default connect()(Login);
