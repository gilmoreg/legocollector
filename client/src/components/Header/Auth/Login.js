import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { login, fetchWatches } from '../../../state/actions';
import './Login.css';

// TODO move to config/env
const API_URL = 'http://localhost:5000';

const storeProfile = ({ token, email }) => {
  console.log('storeProfile', token, email);
  window.localStorage.setItem(
    'legocollectorProfile',
    JSON.stringify({ token, email }),
  );
};

const fetchProfile = accessToken =>
  fetch(`${API_URL}/login`,
    { method: 'POST',
      body: JSON.stringify({ access_token: accessToken }),
    })
    .then(res => res.json())
    .then(res => res.result);

export class Login extends Component {
  constructor(props) {
    super(props);
    this.amazonLogin = this.amazonLogin.bind(this);
  }

  amazonLogin() {
    window.amazon.Login.authorize({ scope: 'profile' },
      response =>
        fetchProfile(response.access_token)
          .then((profile) => {
            storeProfile(profile);
            this.props.dispatch(login(profile));
            this.props.dispatch(fetchWatches(profile.token));
          }));
  }

  render() {
    return (
      <button id="LoginWithAmazon" onClick={this.amazonLogin}>
        <img
          alt="Login with Amazon"
          src="https://images-na.ssl-images-amazon.com/images/G/01/lwa/btnLWA_gold_156x32.png"
          width="156"
          height="32"
        />
      </button>
    );
  }
}

Login.propTypes = {
  dispatch: PropTypes.func.isRequired,
};

export default connect()(Login);
