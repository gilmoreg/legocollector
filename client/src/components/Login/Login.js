import React from 'react';
import './Login.css';

// TODO move to config/env
const API_URL = 'http://localhost:5000';

const fetchProfile = accessToken =>
  fetch(`${API_URL}/login`,
    { method: 'POST',
      body: JSON.stringify({ access_token: accessToken }),
    })
    .then(res => res.json())
    .then(res => console.log(res));

const amazonLogin = () =>
  window.amazon.Login.authorize({ scope: 'profile' },
    response => fetchProfile(response.access_token));

export default () => (
  <button id="LoginWithAmazon" onClick={amazonLogin}>
    <img
      alt="Login with Amazon"
      src="https://images-na.ssl-images-amazon.com/images/G/01/lwa/btnLWA_gold_156x32.png"
      width="156"
      height="32"
    />
  </button>
);

/*
{
  access_token: "Atza|IwEBIEUOpgBKYnfV0HwxYsUDtyI7c4bkSil0tB9kcsLfbP2C_lbPfKs46A9sLS7JHCWHqA_LxvDHI5qHN6XoWyyiuxWlNo0ylbUUGDM3H8pIwebdF0B4_VKUiKFTKIkHXHyFYPy9xM3e2DAhNTaA2XngLHgXCRhKuaoNMDEn8__EjDO6a6azZGEEMcj-7mNxGujj7HxUj0hkuhAFPY_qBsgUdzKgq2I30tdVosGKHBgxzAvGX9nU0Q7lDFj1YWpS3eH6vnu5MfBRPOibjdssC9joAku4KLEZtVO_2M8E5xVMQ1ehmq5K_LfMvqb3JR9j_J2bd3gcJTI80mi5cbru-zOsID0QGa1v8W3NRPhyXT_3EA1Hce_aWgel-PWVc1tddQldUyw_U-4owwIbIKMDJ8-okquXNed5_JZFqeQTGahXZ9pJN5qEYe64XfPl20tG4HzVEXIFYMVLoriOGLk37V3UdLoePDD9SMSw9Co8nbgaiAn60c-fcbcFGxb_2KK3P12KnoCa8dXUIRUG8UXHZq_-XtMjhw66Xz_n52yNbSHBBOJGO9gVeSg8FzJpOX8V77a4pTw"
  scope:"profile"
  status:"complete"
  token_type:"bearer"
}
So with authorization Grant, I would send this token to the server, which would then

*/
