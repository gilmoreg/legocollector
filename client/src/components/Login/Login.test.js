import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Login } from './Login';

configure({ adapter: new Adapter() });

it('renders without crashing', () => {
  const div = document.createElement('div');
  const wrapper = shallow(
    <Login dispatch={()=>{}}/>);
  expect(wrapper).toMatchSnapshot();
});
