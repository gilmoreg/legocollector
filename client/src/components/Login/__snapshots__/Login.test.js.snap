// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    dispatch={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <img
        alt="Login with Amazon"
        height="32"
        src="https://images-na.ssl-images-amazon.com/images/G/01/lwa/btnLWA_gold_156x32.png"
        width="156"
      />,
      "id": "LoginWithAmazon",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "alt": "Login with Amazon",
        "height": "32",
        "src": "https://images-na.ssl-images-amazon.com/images/G/01/lwa/btnLWA_gold_156x32.png",
        "width": "156",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <img
          alt="Login with Amazon"
          height="32"
          src="https://images-na.ssl-images-amazon.com/images/G/01/lwa/btnLWA_gold_156x32.png"
          width="156"
        />,
        "id": "LoginWithAmazon",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "alt": "Login with Amazon",
          "height": "32",
          "src": "https://images-na.ssl-images-amazon.com/images/G/01/lwa/btnLWA_gold_156x32.png",
          "width": "156",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
